### PR TITLE
[sc-134514] Redirect /send requests to push-notifications-2

### PR DESCRIPTION
**Pull request by ahrabovenskyi**
## Description
`POST /send` endpoint is updated to redirect request to `push-notifications-2` if subscription record for the user exists in DB.

## Changes
 - the service connects to new collection `subscriptions` in the same DB (pushserver)
 - function to retrieve a record from `subscriptions` via `sso_id` is added
 - the endpoint is updated to call `push-notifications-2` if subscription record exists for the user in DB
 - if no record exists in DB then old logic is executed
 
 ## Notes
 The update is as minimal as possible since the service will be removed soon

[sc-134514]

### CODE REVIEW

File: lib/PushAssociations.js [hunk 1]
```diff
@@ -29,14 +30,41 @@ var initialize = _.once(function () {
...
+            type: 'String',
+            required: true
+        }
+    });
+
     // I must ensure uniqueness accross the two properties because two users can have the same token (ex: in apn, 1 token === 1 device)
     pushAssociationSchema.index({ user: 1, token: 1 }, { unique: true });
 
     PushAssociation = mongoose.model('PushAssociation', pushAssociationSchema);
+    Subscriptions = mongoose.model('Subscriptions', SubscriptionsSchema);
```
Comment (ahrabovenskyi) [inline lib/PushAssociations.js:52]: connecting to new collection in the DB

File: lib/PushAssociations.js [hunk 2]
```diff
@@ -29,14 +30,41 @@ var initialize = _.once(function () {
...
 
     return module.exports;
 });
 
+const findSubscriptionsForUser = function (ssoId, callback) {
+    Subscriptions.find({sso_id: ssoId}, (err, subscriptions) => {
```
Comment (ahrabovenskyi) [inline lib/PushAssociations.js:58]: retrieves record from `subscriptions` via `sso_id`

File: lib/Web.js [hunk 3]
```diff
@@ -83,13 +84,39 @@ app.post('/unsubscribe', function (req, res) {
...
-    console.log('SENT', req.body.users, notificationsValid);
-
-    res.status(notificationsValid ? 200 : 400).send();
+    /**
+     * users array always has only one user included
+     * we don't send transactional pushes to multiple users in one request
+     */
+    const ssoId = req.body.users[0];
+    PushAssociations.findSubscriptionsForUser(ssoId, (err, subscriptions) => {
+        if (err) {
```
Comment (ahrabovenskyi) [inline lib/Web.js:96]: check if record exists for current user

File: lib/Web.js [hunk 4]
```diff
@@ -83,13 +84,39 @@ app.post('/unsubscribe', function (req, res) {
...
+            console.log(`Error while reading subscriptions from DB for ${ssoId}`, err);
+            res.status(400).send(err.message);
+        } else {
+            if (subscriptions.length > 0) {
+                console.log(`Subscriptions are found. The notification should be sent via OneSignal for ${ssoId}`);
+                
+                oneSignal.postRequestCallbackFunc('send', req.body, (err, result) => {
```
Comment (ahrabovenskyi) [inline lib/Web.js:103]: if record exists then redirect the call to the new service

File: lib/Web.js [hunk 5]
```diff
@@ -83,13 +84,39 @@ app.post('/unsubscribe', function (req, res) {
...
+                    if (err) {
+                        console.log(`Error while sending push notifications for ${ssoId}`)
+                        res.status(400).send(err.message);
+                    } else {
+                        console.log('Sent via OneSignal ', req.body.users, result);
+                        res.status(200).send();
+                    }
+                });
+            } else {
+                console.log('Subscriptions are NOT found. Sending via APN');
```
Comment (ahrabovenskyi) [inline lib/Web.js:113]: if record DOESN'T exist then old logic should be executed

**Review by lalanikki — APPROVED** [review]

**Pull request merged**